services/dqa-secondary-reports/main.py

```diff
@@ -0,0 +1,238 @@
+
+        self.update()
+
+    def process_dqa(self):
+        self.field_totals = {}
+        self.table_totals = {}
+        self.site_totals = {}
+
+        dir_name = './dqa_repo/SecondaryReports/'
+        hospitals = filter(
+            lambda f: (f != 'Ranking'
+                       and os.path.isdir(os.path.join(dir_name, f))),
+            os.listdir(dir_name))
+
+        self.site_totals = {}
+
+        for hospital in hospitals:
+            hosp_path = os.path.join(dir_name, hospital)
+
+            # for each site, we only need data from the latest report for each
+            # data model version, so go through the dqa reports in reverse
+            # order and ignore older reports for each data model version.
+            for etl_version in sorted(os.listdir(hosp_path), reverse=True):
+                etl_path = os.path.join(hosp_path, etl_version)
+
+                if etl_version.startswith('ETL') and os.path.isdir(etl_path):
+                    file_names = list(filter(lambda x: x.endswith('.csv'),
+                                             os.listdir(etl_path)))
+
+                    version = Parser.get_version(
+                        codecs.open(os.path.join(etl_path, file_names[0]), 'r',
+                                    encoding='latin1'))
+
+                    if hospital in self.site_totals.setdefault(version, {}):
+                        continue
+                    else:
+                        self.site_totals[version][hospital] = {}
+
+                    self.table_totals.setdefault(version, {})
+                    self.field_totals.setdefault(version, {})
+
+                    for file_name in file_names:
+                        table = file_name.split('.csv')[0]
+                        buff = codecs.open(
+                            os.path.join(etl_path, file_name), 'r',
+                            encoding='latin1')
+                        p = Parser(buff, hospital, table,
+                                   self.field_totals[version],
+                                   self.table_totals[version],
+                                   self.site_totals[version][hospital])
+                        p.parse()
+
+    def process_dictionary(self):
+        path = './dqa_repo/Dictionary/DCC_DQA_Dictionary.csv'
+        buff = codecs.open(path, 'r', encoding='latin1')
+        self.dictionary = DictionaryParser(buff).parse()
+
+    def update(self):
+        self.update_repo()
+        self.process_dqa()
+        self.process_dictionary()
+
+    def update_repo(self):
+        needs_refresh = True
+
+        if os.path.isdir(self.dir_name):
+            try:
+                git.Repo(self.dir_name).remotes.origin.pull()
+                needs_refresh = False
+            except:
+                shutil.rmtree(self.dir_name)
+
+        if needs_refresh:
+            os.mkdir(self.dir_name)
+            git.Repo.clone_from(self.remote_url, self.dir_name)
+
+    def poll_repo(self, interval):
```

Comment:
Since this is more operational, this should be a standalone function that takes the resource. For example:

```python
def resource_timer(resource, interval):
    if interval < 1:
        interval = 10
    interval *= 60
    
    return threading.Timer(interval, resource.update)

# In main..
# Do the initial fetch of the repository.
dqa_resource.update()
# Create the timer and start it.
timer = resource_timer(dqa_resource, 10)
timer.start()

# Further down when app.run is called..
try:
    app.run(...)
finally:
    timer.cancel()
```